src/verifast1.ml

```diff
@@ -1916,20 +1917,24 @@ module VerifyProgram1(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
               end
             in
             offset_iter fmap (ctxt#mk_intlit 0) true;
-            let base_map = bases |> List.map @@ fun (CxxBaseSpec (l, base, is_virtual)) -> base, (l, is_virtual, get_unique_var_symb (sn ^ "_" ^ base ^ "_offset") intType) in
-            (sn, (l, Some (base_map, fmap), padding_predsym_opt, s))
+            let base_map = bases |> List.map @@ fun (CxxBaseSpec (l, base, is_virtual)) -> 
+              let base_offset = get_unique_var_symb (sn ^ "_" ^ base ^ "_offset") intType in
+              ctxt#assert_term (ctxt#mk_le int_zero_term base_offset);
+              base, (l, is_virtual, base_offset) 
+            in
+            (sn, (l, Some (base_map, fmap), padding_predsym_opt, s, type_info))
           | Field (lf, gh, t, f, Instance, Public, final, init)::fds ->
             if List.mem_assoc f fmap then static_error lf "Duplicate field name." None;
             let t = check_pure_type ("", []) [] gh t in
             let offset = if gh = Ghost then None else Some (get_unique_var_symb (sn ^ "_" ^ f ^ "_offset") intType) in
             let entry = (f, (lf, gh, t, offset, init)) in
-            iter1 (entry::fmap) fds (has_ghost_fields || gh = Ghost) bases
+            iter1 (entry::fmap) fds (has_ghost_fields || gh = Ghost) bases type_info
         in
-        let new_item = begin
+        let new_item = 
+          let type_info = get_unique_var_symb (sn ^ "_type_info") type_info_ref_type in

```

Comment:
Move this line before the definition of `iter1` so you don't need the extra `iter1` parameter?